Automattic/wp-calypso
# Remove the isSectionIsomorphic selector

The `isSectionIsomorphic` selector is used only by the SSR code on the server. This PR starts setting `context.section` in the routing code, in addition to dispatching the `setSection` Redux action, and therefore the `section` object is available to the router code in the router `context`.

As a result, the routing code no longer needs to read stuff from Redux at all. Redux state is needed only by React UI when rendering. The `isSectionIsomorphic` selector can be removed.

This PR also contains two commits that merely rename routing middlewares (`setLocaleMiddleware`, `setSectionMiddleware`) and import aliases for Redux action (`setRoute`, `setSection`). The purpose is to give things consistent names that don't conflict with each other and that can be imported without aliasing.

Best reviewed commit-by-commit -- all commits are focused on one thing and are fairly independent from each other.

**How to test:**
Verify that SSR (for login and for themes) works as expected under all conditions: enabled for logged out users and `en` locale, disabled for logged-in and non-`en`.
